py/objdict.c

```diff
@@ -70,7 +74,14 @@ STATIC void dict_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_
             mp_print_str(print, ", ");
         }
         first = false;
+	bool quote = quote_nums && (mp_obj_is_int(next->key) || mp_obj_is_float(next->key));
```

Comment:
To cover the case of True/False/None it's probably better to have this as a check for strings, and if not a string then add quotes.  That would also be more efficient, only a single check.

Reply:
Pushed update, per your suggestions.